src/components/Login.vue

```diff
@@ -0,0 +1,116 @@
+<script setup lang="ts">
+import { computed, ref } from 'vue'
+import { Account } from '../lib/api/Account'
+import {
+	LoginRequest,
+	ValidationProblemDetails
+} from '../lib/api/data-contracts'
+import { useSessionToken } from '../composables/useSessionToken'
+import { useUserDetails } from '../composables/useUserDetails'
+import { useApi } from '../composables/useApi'
+
+const email = ref('')
+const password = ref('')
+const errorText = ref('')
+const submitted = ref(false)
+const token = useSessionToken()
+const user = useUserDetails()
+const nonAdminLogin = computed(
+	() =>
+		submitted.value &&
+		token.value &&
+		user.value &&
+		user.value.role !== 'Administrator'
+)
+
+async function submit() {
+	const request: LoginRequest = {
+		email: email.value,
+		password: password.value
+	}
+
+	await useApi(
+		() =>
+			new Account({
+				baseUrl: import.meta.env.VITE_BACKEND_URL
+			}).login(request),
+		(success) => {
+			token.value = success.data.token
+			submitted.value = true
+		},
+		(error) => {
+			submitted.value = false
+
+			switch (error.status) {
+				case 401:
+					errorText.value = 'Incorrect password'
+					break
+
+				case 403:
+					errorText.value = 'User banned'
+					break
+
+				case 404:
+					errorText.value = 'User not found'
+					break
+
+				case 422:
+					const validationProblemDetails =
+						error.error as ValidationProblemDetails
+					errorText.value = validationProblemDetails.title || ''
+					break
+
+				case 500:
+					errorText.value = 'Server error'
+					break
+
+				default:
+					errorText.value = 'Unknown error'
```

Comment:
I guess it's kind of unnecessary and we're gonna change it on the backend so that it doesn't give this info and just gives either success, banned, or failure. Do you want me to change it to just say "login failed" or something for all cases?

Reply:
I'm good with a simple "login failed"